src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java

```diff
@@ -466,8 +467,33 @@ public OffsetAndMetadata committed(TopicPartition topicPartition, Duration durat
   @Override
   public Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes(
       Map<TopicPartition, Long> map, Duration duration) {
-    throw new UnsupportedVersionException(
-        "Pub/Sub Lite does not support Consumer backlog introspection.");
+    try {
+      Map<TopicPartition, ApiFuture<Optional<Cursor>>> cursors =
+          map.entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      entry -> entry.getKey(),
+                      entry ->
+                          topicStatsClient.computeCursorForEventTime(
+                              topicPath,
+                              checkTopicGetPartition(entry.getKey()),
+                              Timestamps.fromMillis(entry.getValue()))));
+      ApiFutures.allAsList(cursors.values()).get(duration.toMillis(), TimeUnit.MILLISECONDS);
+
+      // As per KafkaConsumer.offsetsForTimes, null (represented by no map entry) is returned if
```

Comment:
this is not the correct behavior. What the api should return is a mapping from the topic partition to the literal null.

Reply:
Fixed. Now returns a HashMap, which permits null values.